project_generation/content/templates/api/main.go.tmpl

```diff
@@ -19,23 +20,42 @@ var (
 	GitCommit string
 	// Version represents the version of the service that is running
 	Version string
+
+/* NOTE: replace the above with the below to run code with for example vscode debugger.
+BuildTime string = "1601119818"
+GitCommit string = "6584b786caac36b6214ffe04bf62f058d4021538"
+Version   string = "v0.1.0"
+*/
 )
 
 func main() {
 	log.Namespace = serviceName
+	ctx := context.Background()
 
-	if err := run(); err != nil {
+	if err := run(ctx); err != nil {
 		log.Event(nil, "fatal runtime error", log.Error(err), log.FATAL)
 		os.Exit(1)
 	}
 }
 
-func run() error {
+func run(ctx context.Context) error {
 	signals := make(chan os.Signal, 1)
 	signal.Notify(signals, os.Interrupt, os.Kill)
 
+	// Run the service, providing an error channel for fatal errors
 	svcErrors := make(chan error, 1)
-	svc, err := service.Run(BuildTime, GitCommit, Version, svcErrors)
+	svcList := service.NewServiceList(&service.Init{})
```

Comment:
Just an observation, maybe for a future task: I think that the service initialiser may be confusing, and we should follow `dp-filter-api` as an example. But I wouldn't block this task for this reason.

Reply:
leave this one for another PR (maybe)